### PR TITLE
[NFC] Eliminate top-level "using namespace" from some headers.

### PR DESCRIPTION
**Pull request by jurahul**
- Eliminate top-level "using namespace" from some headers.

### CODE REVIEW

File: llvm/lib/ExecutionEngine/RuntimeDyld/RuntimeDyldCOFF.h [hunk 1]
```diff
@@ -15,10 +15,6 @@
 
 #include "RuntimeDyldImpl.h"
 
-#define DEBUG_TYPE "dyld"
```
Comment (jurahul) [inline llvm/lib/ExecutionEngine/RuntimeDyld/RuntimeDyldCOFF.h:18]: This header code does not seems to use anything like LLVM_DEBUG which needs the DEBUG_TYPE to be set.

File: llvm/lib/ExecutionEngine/RuntimeDyld/RuntimeDyldMachO.h [hunk 2]
```diff
@@ -17,11 +17,6 @@
 #include "llvm/Object/MachO.h"
 #include "llvm/Support/Format.h"
 
-#define DEBUG_TYPE "dyld"
```
Comment (jurahul) [inline llvm/lib/ExecutionEngine/RuntimeDyld/RuntimeDyldMachO.h:20]: Same observation here as well, this DEBUG_TYPE define is unsed.

File: llvm/lib/ExecutionEngine/RuntimeDyld/Targets/RuntimeDyldCOFFAArch64.h [hunk 3]
```diff
@@ -22,8 +21,6 @@
 
 #define DEBUG_TYPE "dyld"
```
Comment (jurahul) [inline llvm/lib/ExecutionEngine/RuntimeDyld/Targets/RuntimeDyldCOFFAArch64.h:22]: Added undef for this one (which seems to be used) at the end of the header.

File: llvm/lib/Transforms/InstCombine/InstCombineNegator.cpp [hunk 4]
```diff
@@ -46,12 +46,8 @@
 #include <type_traits>
 #include <utility>
 
-namespace llvm {
```
Comment (jurahul) [inline llvm/lib/Transforms/InstCombine/InstCombineNegator.cpp:49]: These declarations seems unnecessary as they are already provided by the included headers.

File: llvm/lib/Target/RISCV/MCTargetDesc/RISCVELFStreamer.h [hunk 5]
```diff
@@ -12,7 +12,7 @@
 #include "RISCVTargetStreamer.h"
 #include "llvm/MC/MCELFStreamer.h"
 
-using namespace llvm;
+namespace llvm {
```
Comment (nikic) [inline llvm/lib/Target/RISCV/MCTargetDesc/RISCVELFStreamer.h:15]: There's a `namespace llvm {` below that you should move instead.

Comment (jurahul) [reply llvm/lib/Target/RISCV/MCTargetDesc/RISCVELFStreamer.h:15]: Right, that's why the build is failing. Let me make sure all the backends this is touching build locally for me before making it ready for review again.

File: llvm/lib/ExecutionEngine/RuntimeDyld/RuntimeDyldCOFF.h (hunk 1 above)
Comment (s-barannikov) [reply llvm/lib/ExecutionEngine/RuntimeDyld/RuntimeDyldCOFF.h:18]: This is probably not worth touching. See https://discourse.llvm.org/t/rfc-removing-mcjit-and-runtimedyld/80464

**Comment by jurahul** [comment]
All checks are clean now. @nikic can you please review?

File: llvm/lib/ExecutionEngine/RuntimeDyld/RuntimeDyldCOFF.h (hunk 1 above)
Comment (jurahul) [reply llvm/lib/ExecutionEngine/RuntimeDyld/RuntimeDyldCOFF.h:18]: Thanks. Taking a cursory look, it seems it's not yet decided when the code will be removed. As such, might to ok to commit it. But if there is a push-back, I can undo the changes in some files.

**Comment by jurahul** [comment]
Also, I wonder if there is a clang-tidy warning for this. I'll try to find.

**Review by nikic — APPROVED** [review]
LGTM

**Comment by jurahul** [comment]
Thanks. Can one of you please merge the PR?

**Comment by firewave** [comment]
> Also, I wonder if there is a clang-tidy warning for this. I'll try to find.

There is `google-build-using-namespace`: https://clang.llvm.org/extra/clang-tidy/checks/google/build-using-namespace.html. That flags *all* usages though. It (or an alias) should probably get an option to limit it to headers only.

**Pull request merged**